### PR TITLE
Fixed #3596: counts for entities using list APIs by setting limit=0

### LINKED ISSUE #3596: Provide option to get counts for entities using list APIs  by setting limit=0
**Affected module**
Does it impact the UI, backend or Ingestion Framework?
UI. 

**Describe the bug**
Services count shown is 13 whereas count for all the services is 30. Looks like ElasticSearch is aggregating a wrong value from the query.

**To Reproduce**

Ingest lot of services and then check for count on HomePage of OpenMetadata UI.
Also, tried to run metadata_to_es.json ingestion which reindexes Elastic Search. Post that, the count did not change (Ideally this should not be required with the change event handler).

image
image
image

**Expected behavior**
UI HomePage should show count of all the services ingested in open metadata.

**Version:**
 - OS: Ubuntu

cc - @harshach , @sureshms

### PR DESCRIPTION
**Pull request by darth-coder00**
### Describe your changes :
Closes #3596 

I worked on the REST api integration for assets count on landing page

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :
<p align="center">
Screenshot 2022-03-24 at 9 15 22 PM

</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@harshach @vivekratnavel @Sachin-chaurasiya

### CODE REVIEW

**Review by vivekratnavel — CHANGES_REQUESTED** [review]

**Review by harshach — CHANGES_REQUESTED** [review]
@darth-coder00 this test is not good enough. We are only testing with happy paths.
What if the API didn't return paging as a field in there do we fail and show a blank page?
What would be defaults if the counts didn't come through and what if the API response is changed can we add those test cases

File: openmetadata-ui/src/main/resources/ui/src/interface/service.interface.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  server?: string;
+  env?: string;
+}
+
+export interface DataObj {
+  id?: string;
+  description: string | undefined;
+  ingestionSchedule?:
+    | {
+        repeatFrequency: string;
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/interface/service.interface.ts:42]: Can we make this a type and reuse it everywhere?

File: openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx [hunk 2]
```diff
@@ -71,31 +65,63 @@ const MyDataPage = () => {
...
-            getEntityCountByType(
-              res.data.aggregations?.['sterms#EntityType']?.buckets
-            )
-          );
+    getAllTables('', 0)
+      .then((res) => {
+        if (res.data) {
+          setCountTables(res.data.paging.total || 0);
+        } else {
+          throw '';
```
[the remark this answers is not in the document]
Comment (vivekratnavel) [reply openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx:85]: This conditional still needs to be handled.

File: openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx [hunk 3]
```diff
@@ -88,10 +102,15 @@ const MyDataPage = () => {
           throw '';
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx:102]: Handle this condition in a better way.

File: openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx [hunk 4]
```diff
@@ -100,10 +119,15 @@ const MyDataPage = () => {
...
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx:119]: Handle this condition in a better way.

File: openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx [hunk 5]
```diff
@@ -249,6 +285,26 @@ const MyDataPage = () => {
       });
   };
 
+  const fetchOMDMode = () => {
+    fetchSandboxConfig()
+      .then((res) => {
+        if (res.data) {
+          setIsSandbox(Boolean(res.data.sandboxModeEnabled));
+        } else {
+          throw '';
```
Comment (vivekratnavel) [inline openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx:294]: Show unexpected response from the server.

**Review by vivekratnavel — CHANGES_REQUESTED** [review]

File: openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx (hunk 5 above)
Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx:294]: Same error is there as fallback text in catch block, so did not pass it explicitly with `throw`. Will add it.

Comment (darth-coder00) [reply openmetadata-ui/src/main/resources/ui/src/pages/MyDataPage/MyDataPage.component.tsx]: Addressed

**Review by vivekratnavel — APPROVED** [review]

**Pull request merged**